### ISSUE TITLE
HD doesn't recognize the checked attr in the input tag

### ISSUE THREAD

**Issue by aslanalyiev** [capricorn86/happy-dom#631]
**Describe the bug**
Not working with input checked={isChecked} prop in a react proj.

**To Reproduce**
```
const Toggle = ({
  value,
  checked = false,
  onChange,
  text
}: Props) => {
  const [checked, setChecked] = useState(false);

  const onToggle = () => {
    if (!disabled) {
      if (onChange) onChange();
      setChecked(prev => !prev);
    }
  };

  return (
    <div className="toggle" data-testid="toggle">
      <label data-testid="toggle-label" className="toggle-switch">
        <input
          data-testid="toggle-input"
          value={value}
          type="checkbox"
          checked={checked}
          onChange={onToggle}
        />
      </label>
    </div>
  );
};
```

**Expected behavior**
**PASS THIS TEST:** 
```
render(<Toggle onChange={handleChange} text={someText} value={someVal} />);

    expect(screen.getByTestId("toggle")).toBeInTheDocument();
    expect(screen.getByTestId("toggle-input")).not.toBeChecked();

    await userEvent.click(screen.getByTestId("toggle-label"));

    expect(handleChange).toBeCalledTimes(1); //FAILS
    expect(screen.getByTestId("toggle-input")).toBeChecked();  //FAILS
```

**Screenshots**

**Device:**
 - OS: macOs 12.6
 - Browser [chrome]
 - Version [106]

**Additional context**
I tried to use vitest with HD, but it didn't work until I switched to jsdom

**Comment by christianlmc** [comment]
Had that same issue on Vue 3, switching to jsdom fixed my issue as well

**Comment by aslanalyiev** [comment]
@christianlmc 

Yeah, I know but it is strange that happy-dom doesn't support it :)

**Comment by dspinov** [comment]
I have the same problem. I'd like to see any updates on this issue as I also had to switch back to jsdom because of this.

**Comment by mehdi-valette** [comment]
Hello,

I noticed that the checkbox doesn't change its checked value and doesn't trigger onChange, that's why the test fails.

Adding the following function inside the file `nodes/html-input-element/HTMLInputElement.js` fixed the issue for me:

```javascript
onclick = function() {
		switch (this.type) {
			case 'checkbox':
				if (this.checked) {
					this.removeAttributeNS(null, 'checked');
				}
				else {
					this.setAttributeNS(null, 'checked', '');
				}
		}
	}
```

Note that I used a "switch" because ideally it should handle at least two other cases:
- type `radio` should set the attribute `checked` to `true` if they weren't checked, and uncheck the previously checked radio button from the same Form with the same name.
- type `submit` should trigger the event onSubmit of the parent form.

In case you're wondering, it wasn't necessary to trigger onChange because apparently changing the attribute will automatically trigger onChange. That may be a problem in the `radio` case because a `radio` that becomes unchecked should not trigger onChange.

I could open a pull request for these cases, but I just discovered happy-dom so I may not be the best person for the task.
